src/wallet/rpc/backup.cpp

```diff
@@ -247,7 +250,15 @@ RPCHelpMan importaddress()
     std::shared_ptr<CWallet> const pwallet = GetWalletForJSONRPCRequest(request);
     if (!pwallet) return UniValue::VNULL;
 
-    EnsureLegacyScriptPubKeyMan(*pwallet, true);
+    bool has_legacy = pwallet->GetLegacyScriptPubKeyMan();
+    if (!has_legacy) {
+        // As we don't allow mixing watch-only descriptors with spendable ones,
+        // continue creating a legacy spkm if that is the case
```

Comment:
Let's not create a legacy spkm inside a descriptor wallet. Just fail instead, and tell the user to use `importdescriptors` for that use case.

Reply:
This is actually to retain the current behavior, it does not create an hybrid wallet; the legacy spkm creation only happens on blank non-descriptors wallet.

The internal `EnsureLegacyScriptPubKeyMan` workflow uses `SetupLegacyScriptPubKeyMan` which fails if the wallet contains any spkm or the `WALLET_FLAG_DESCRIPTORS` wallet flag.

Reply:
Ah, that's worth mentioning in the comment, and maybe also covering with a test.

Reply:
yeah 👍🏼

Reply:
Pushed test coverage for it.